src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -548,33 +569,21 @@ private void replaceEC2Servers() {
 
                 // Check if a new image of the instance with the completed graph build should be created.
                 if (otpServer.ec2Info.recreateBuildImage) {
-                    status.update("Creating build image", 42.5);
-                    // Create a new image of this instance.
-                    CreateImageRequest createImageRequest = new CreateImageRequest()
-                        .withInstanceId(graphBuildingInstances.get(0).getInstanceId())
-                        .withName(otpServer.ec2Info.buildImageName)
-                        .withDescription(otpServer.ec2Info.buildImageDescription);
-                    CreateImageResult createImageResult = ec2.createImage(createImageRequest);
-                    // Deregister old image if it exists
-                    if (otpServer.ec2Info.buildAmiId != null) {
-                        status.message = "Deregistering old build image";
-                        DeregisterImageRequest deregisterImageRequest = new DeregisterImageRequest()
-                            .withImageId(otpServer.ec2Info.buildAmiId);
-                        ec2.deregisterImage(deregisterImageRequest);
-                    }
-                    status.update("Updating Server build AMI info", 45);
-                    // Update OTP Server info
-                    otpServer.ec2Info.buildAmiId = createImageResult.getImageId();
-                    otpServer.ec2Info.recreateBuildImage = false;
-                    Persistence.servers.replace(otpServer.id, otpServer);
+                    // Begin a new job to create a graph build image so that can happen asynchronously
+                    RecreateBuildImageJob recreateBuildImageJob = new RecreateBuildImageJob(
```

Comment:
This will only happen in a reasonable amount of time if there are enough threads available in the heavyExecutor. If not, it could get queued up for a very long time. I'm not sure we want to do this...

Reply:
Hmmm. This makes me think we ought to have a 3rd type of executor for long-lived, but low CPU tasks and then we would put all of these deploy job tasks that are basically only making HTTP requests.

Reply:
I think a deploy job could still be high in memory requirements - e.g., if it's using build graph over wire. Why not just have this RecreateBuildImageJob run in the lightExecutor? Should the lightExecutor have more threads at its disposal? I'm just not aware of all of thread safety concerns if we increase the thread count (insert Egyptian cotton joke here).